src/libs/PDCurses/sdl2_queue/pdckbd.cpp

```diff
@@ -81,17 +81,23 @@ void PDC_set_keyboard_binary(bool on)
 
 bool PDC_check_key(void)
 {
-    int haveevent;
+    int haveevent = 0;
 
     PDC_pump_and_peep();
 
     /* SDL_TEXTINPUT can return multiple chars from the IME which we
        should handle before polling for additional events. */
 
-    if (event.type == SDL_TEXTINPUT && event.text.text[0])
+    if (event.type == SDL_TEXTINPUT && event.text.text[0]) {
         haveevent = 1;
-    else
-        haveevent = SDL_PollEvent(&event);
+    } else {
+        haveevent = !pdc_event_queue.empty();
+        if (haveevent) {
+            const auto ev = pdc_event_queue.front();
+            event = ev;
```

Comment:
(similar about double-negative and reference).

Is it possible to assign straight into `event = pdc_event_queue.front();`?

Reply:
Here I don't think we can, because we need a copy of it for their global `event`, and we'll lose the reference to it after the subsequent `pop`?

Reply:
> (similar about double-negative and reference).

I avoided re-doing the logic of `haveevent` here, because you have an event if the queue is not empty.
 
> Is it possible to assign straight into `event = pdc_event_queue.front();`?

👍